daml-lf/spec/daml-lf-1.rst

```diff
@@ -643,6 +645,11 @@ Then we can define our kinds, types, and expressions::
        |  'TypeRep'                                 -- BTTypeRep
        |  'Update'                                  -- BTyUpdate
        |  'Scenario'                                -- BTyScenario
+       -- [*Available in version >= 1.dev*]
+       |  'AnyException'                            -- BTyAnyException
+       |  'GeneralError'                            -- BTyGeneralError
+       |  'ArithmeticError'                         -- BTyArithmeticError
+       |  'ContractError'                           -- BTyContractError
```

Comment:
I see you version the expression but not the type checking rule. 
I am fine with this approach, but coherence is more important for me. 
Either you follow the convention which are already in place (you do not version the expression but version the type checking rule as it is done for `Any` and `TypeRep`) or you change how think are version to your new convention.

Second point, I would be very please if you version each expression independently, i.e. something like that: 
```
       |  'AnyException'                            -- BTyAnyException    [DAML-LF ≥ 1.dev]
       |  'GeneralError'                            -- BTyGeneralError    [DAML-LF ≥ 1.dev]
       |  'ArithmeticError'                         -- BTyArithmeticError [DAML-LF ≥ 1.dev]
       |  'ContractError'                           -- BTyContractError   [DAML-LF ≥ 1.dev]
```
This scheme is simpler (and safer) to update when we release the new version of LF.

Reply:
I couldn't fine any precendence for having type checking rules that depend on the DAML-LF version (the `Any` stuff doesn't do it) but I think it makes to have it there. I'm moving it over.

Reply:
You are right, there is no versioning for `Any` --it is an issue--, I have been to quick.
The original idea, was to not version anything in the definition and the inference rule but just add a paragraph in the `Serialization changes` section. See for instance `Generic Map`.